src/main/java/com/example/airportproject/controller/FlightController.java

```diff
@@ -1,23 +1,62 @@
 package com.example.airportproject.controller;
 
-import com.example.airportproject.domain.Flight;
+import com.example.airportproject.model.Flight;
+import com.example.airportproject.exception.FlightNotFoundException;
 import com.example.airportproject.repository.FlightRepository;
+import com.example.airportproject.service.FlightService;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
 @RestController
+@RequestMapping("api/flights")
 public class FlightController {
-    //private FlightService
+    private final FlightService flightService;
+    private final FlightRepository flightRepository;
 
-    private FlightRepository flightRepository;
+    @Autowired
+    public FlightController(FlightRepository flightRepository, FlightService flightService) {
+        this.flightRepository = flightRepository;
+        this.flightService = flightService;
 
-    @GetMapping("/getAll")
-    public List<Flight> getUsers() {
+    }
+
+    @GetMapping("/")
+    public List<Flight> getFlights() {
         return flightRepository.findAll();
     }
 
-    // other controller methods
+    @GetMapping("/{id}")
+    public Flight getFlightById(@PathVariable int id){
+        return flightRepository.findById((long) id).orElseThrow(FlightNotFoundException::new);
+    }
+
+    @PostMapping("/create")
+    public ResponseEntity<Flight> createFlight(@RequestBody Flight f){
+        Flight created = this.flightService.createFlight(f);
+        return new ResponseEntity<>(created, HttpStatus.CREATED);
```

Comment:
Use the builder methods for your response entities, and have a look into ResponseEntity.created() for including the location of a created entry.
You can also look at exposing the location header so that the client has access to it.